pkg/factory/risingwave_object_factory.go

```diff
@@ -843,6 +843,28 @@ func mergeComponentGroupTemplates(base, overlay *risingwavev1alpha1.RisingWaveCo
 	}
 	r.NodeSelector = mergeMap(r.NodeSelector, overlay.NodeSelector)
 	r.PodTemplate = mergeValue(r.PodTemplate, overlay.PodTemplate)
+	if len(overlay.Tolerations) != 0 {
+		r.Tolerations = overlay.Tolerations
+	}
+	r.PriorityClassName = mergeValue(r.PriorityClassName, overlay.PriorityClassName)
+	if overlay.SecurityContext != nil {
+		r.SecurityContext = overlay.SecurityContext.DeepCopy()
+	}
+	if overlay.DNSConfig != nil {
+		r.DNSConfig = overlay.DNSConfig.DeepCopy()
+	}
+	if overlay.TerminationGracePeriodSeconds != nil {
+		r.TerminationGracePeriodSeconds = pointer.Int64(*overlay.TerminationGracePeriodSeconds)
+	}
```

Comment:
Could we simplify this using `mergeValue`?

```suggestion
	r.SecurityContext = mergeValue(overlay.SecurityContext, nil).DeepCopy()
	r.DNSConfig = mergeValue(overlay.DNSConfig, nil).DeepCopy()
	r.TerminationGracePeriodSeconds = pointer.Int64(*mergeValue(overlay.TerminationGracePeriodSeconds, nil))
```